brave/brave6/src/main/java/com/linecorp/armeria/server/brave/BraveRpcService.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2025 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.brave;
+
+import static com.linecorp.armeria.internal.common.brave.TraceContextUtil.ensureScopeUsesRequestContext;
+import static com.linecorp.armeria.server.brave.ArmeriaServerParser.annotateWireSpan;
+import static com.linecorp.armeria.server.brave.BraveService.SERVICE_REQUEST_DECORATING_SCOPE;
+
+import java.util.function.Function;
+
+import com.linecorp.armeria.common.RpcRequest;
+import com.linecorp.armeria.common.RpcResponse;
+import com.linecorp.armeria.common.brave.RequestContextCurrentTraceContext;
+import com.linecorp.armeria.internal.common.RequestContextExtension;
+import com.linecorp.armeria.server.RpcService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.SimpleDecoratingRpcService;
+import com.linecorp.armeria.server.TransientServiceOption;
+
+import brave.Span;
+import brave.Tracer;
+import brave.Tracer.SpanInScope;
+import brave.Tracing;
+import brave.rpc.RpcRequestParser;
+import brave.rpc.RpcResponseParser;
+import brave.rpc.RpcServerHandler;
+import brave.rpc.RpcServerRequest;
+import brave.rpc.RpcServerResponse;
+import brave.rpc.RpcTracing;
+
+/**
+ * Decorates an {@link RpcService} to trace inbound {@link RpcRequest}s using
+ * <a href="https://github.com/openzipkin/brave">Brave</a>.
+ */
+public final class BraveRpcService extends SimpleDecoratingRpcService {
```

Comment:
Question) Is it possible to share the common logic with `BraveService` by adding `AbstractBraveService` and extending it, as you did in `AbstractMetricCollectingClient`?